app/controllers/applications_controller.rb

```diff
@@ -19,14 +19,20 @@ def update
     ApplicationPet.create!(pet: selected_pet, application: application)
     redirect_to "/applications/#{params[:application_id]}"
   end
-  
+
   def new
   end
 
   def create
     params[:address] = "#{params[:street_address]}, #{params[:city]}, #{params[:state]}, #{params[:zip_code]}"
-    app = Application.create!(application_params)
-    redirect_to "/applications/#{app.id}"
+    app = Application.new(application_params)
+    if app.save
+      redirect_to "/applications/#{app.id}"
+      return
```

Comment:
I don't think you need a return here because you have the redirect on the line above, did you try testing this without a return on line 31?